File: Daily Exercises/Mar 2021/2021-03-25 : 82. Remove Duplicates from Sorted List II/82. Remove Duplicates from Sorted List II.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* p=head;
        if(p==nullptr || p->next==nullptr){
            return p;
        }
        if(p->val==p->next->val){
            while(p!=nullptr && p->val==head->val){
                p=p->next;
            }
            return deleteDuplicates(p);
        }else{
            p->next=deleteDuplicates(p->next);
            return p;
        }
    }
};

File: Daily Exercises/Mar 2021/2021-03-25 : 82. Remove Duplicates from Sorted List II/82. Remove Duplicates from Sorted List II.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode p=head;
        if(p==null || p.next==null){
            return p;
        }
        if(p.val==p.next.val){
            while(p!=null && p.val==head.val){
                p=p.next;
            }
            return deleteDuplicates(p);
        }else{
            p.next=deleteDuplicates(p.next);
            return p;
        }
    }
}
